### ISSUE TITLE
faster-rcnn-resnet50-fpn mutil batchsize can support ？ and why now cannot mutil batchsize 

### ISSUE THREAD

**Issue by zyxcambridge** [tensorpack/tensorpack#1344]
faster-rcnn-resnet50-fpn mutil batchsize can support ？ and why now cannot mutil batchsize

**Comment by zyxcambridge** [comment]
def preprocess(self, image):
        image = tf.expand_dims(image, 0)
        image = image_preprocess(image, bgr=True)
        return tf.transpose(image, [0, 3, 1, 2])

**Comment by ppwwyyxx** [comment]
answered in https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md

**Comment by zyxcambridge** [comment]
> answered in https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md

Possible Future Enhancements:

Support batch>1 per GPU. Batching with inconsistent shapes is non-trivial to implement in TensorFlow.

Use dedicated ops to improve speed. (e.g. a TF implementation of ROIAlign op can be found in light-head RCNN)


给个建议 怎么修改 batchsize 》 1

**Comment by ppwwyyxx** [comment]
As it said, we do not support batch size > 1.

**Comment by zyxcambridge** [comment]
> As it said, we do not support batch size > 1.

我目前的输入图片尺寸 是固定的 ，都是 600*800；要支持多batchsize训练，直接稍微改一下一应就可以了吧；
pytorch 如此处理：

class ImageList(object):
    """
    Structure that holds a list of images (of possibly
    varying sizes) as a single tensor.
    This works by padding the images to the same size,
    and storing in a field the original sizes of each image
    """

    def __init__(self, tensors, image_sizes):
        """
        Arguments:
            tensors (tensor)
            image_sizes (list[tuple[int, int]])
        """
        self.tensors = tensors
        self.image_sizes = image_sizes

    def to(self, *args, **kwargs):
        cast_tensor = self.tensors.to(*args, **kwargs)
        return ImageList(cast_tensor, self.image_sizes)



那么对于 tf 给点建议

**Comment by ppwwyyxx** [comment]
> class ImageList(object):

It takes much more code than this class to handle batches in pytorch as well.

We do not handle batches in this implementation. There is a fork of this implementation that handles batches, implemented for a subset of models: https://github.com/armandmcqueen/tensorpack-mask-rcnn
